lib/spack/spack/install_test.py

```diff
@@ -52,6 +93,638 @@ def get_test_stage_dir():
     )
 
 
+def cache_extra_test_sources(pkg: "spack.package_base.PackageBase", srcs: ListOrStringType):
+    """Copy relative source paths to the corresponding install test subdir
+
+    This routine is intended as an optional install test setup helper for
+    grabbing source files/directories during the installation process and
+    copying them to the installation test subdirectory for subsequent use
+    during install testing.
+
+    Args:
+        pkg: package being tested
+        srcs: relative path for file(s) and or subdirectory(ies) located in
+            the staged source path that are to be copied to the corresponding
+            location(s) under the install testing directory.
+    """
+    paths = [srcs] if isinstance(srcs, str) else srcs
+
+    for path in paths:
+        src_path = os.path.join(pkg.stage.source_path, path)
+        dest_path = os.path.join(install_test_root(pkg), path)
+        if os.path.isdir(src_path):
+            fs.install_tree(src_path, dest_path)
+        else:
+            fs.mkdirp(os.path.dirname(dest_path))
+            fs.copy(src_path, dest_path)
+
+
+def check_outputs(expected: Union[list, set, str], actual: str):
+    """Ensure the expected outputs are contained in the actual outputs.
+
+    Args:
+        expected: expected raw output string(s)
+        actual: actual output string
+
+    Raises:
+        RuntimeError: the expected output is not found in the actual output
+    """
+    expected = expected if isinstance(expected, (list, set)) else [expected]
+    for check in expected:
+        if not re.search(check, actual):
+            raise RuntimeError("Expected '{0}' in output '{1}'".format(check, actual))
+
+
+def find_required_file(
+    root: str, filename: str, expected: int = 1, recursive: bool = True
+) -> ListOrStringType:
+    """Find the required file(s) under the root directory.
+
+    Args:
+       root: root directory for the search
+       filename: name of the file being located
+       expected: expected number of files to be found under the directory
+           (default is 1)
+       recursive: ``True`` if subdirectories are to be recursively searched,
+           else ``False`` (default is ``True``)
+
+    Returns: the path(s), relative to root, to the required file(s)
+
+    Raises:
+        Exception: SkipTest when number of files detected does not match expected
+    """
+    paths = fs.find(root, filename, recursive=recursive)
+    num_paths = len(paths)
+    if num_paths != expected:
+        files = ": {0}".format(", ".join(paths)) if num_paths else ""
+        raise SkipTest(
+            "Expected {0} of {1} under {2} but {3} found{4}".format(
+                plural(expected, "copy", "copies"),
+                filename,
+                root,
+                plural(num_paths, "copy", "copies"),
+                files,
+            )
+        )
```

Comment:
Are we using this exception to signal that a test is skipped? In that case wouldn't it be better to not derive from `SpackError`? I'm thinking of code written like:
```python
try:
   # <some call including the above>
except SpackError as e:
   raise AnotherError("msg") from e

```

Reply:
Yes, to show a test is skipped.  Excellent point on what exception type to derive from.  Thanks!

Reply:
Done.